android-huawei/src/de/golfgl/gdxgamesvcs/HuaweiGameServicesClient.java

```diff
@@ -0,0 +1,631 @@
+        Task<OperationResult> addArchiveTask = this.archivesClient.loadArchiveDetails(fileId);
+        addArchiveTask.addOnSuccessListener(new OnSuccessListener<OperationResult>() {
+            @Override
+            public void onSuccess(OperationResult result) {
+                Archive archive = result.getArchive();
+
+                try {
+                    byte[] data = archive.getDetails().get();
+                    responseListener.gsGameStateLoaded(data);
+                } catch (IOException e) {
+                    sendError(IGameServiceListener.GsErrorType.errorUnknown, e.getMessage(), e);
+                }
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(Exception e) {
+                sendError(IGameServiceListener.GsErrorType.errorUnknown, e.getMessage(), e);
+            }
+        });
+    }
+
+    @Override
+    public boolean deleteGameState(String fileId, final ISaveGameStateResponseListener successCallback) {
+        if (!this.isSaveDataEnabled) {
+            return false;
+        }
+
+        Task<List<ArchiveSummary>> task = this.archivesClient.getArchiveSummaryList(true);
+        task.addOnSuccessListener(new OnSuccessListener<List<ArchiveSummary>>() {
+            @Override
+            public void onSuccess(List<ArchiveSummary> archiveSummaries) {
+                ArchiveSummary firstSaveData = archiveSummaries != null && !archiveSummaries.isEmpty() ? archiveSummaries.get(0) : null;
+
+                if (firstSaveData != null) {
+                    deleteSaveGame(firstSaveData, successCallback);
+                } else if (successCallback != null){
+                    successCallback.onGameStateSaved(true, null);
+                }
+            }
+        }).addOnFailureListener(new OnFailureListener() {
+            @Override
+            public void onFailure(Exception e) {
+                sendError(IGameServiceListener.GsErrorType.errorUnknown, e.getMessage(), e);
+            }
+        });
+
+        return true;
+    }
+
+    private void deleteSaveGame(ArchiveSummary archiveSummary, final ISaveGameStateResponseListener successCallback) {
+        Task<String> task = this.archivesClient.removeArchive(archiveSummary);
+
+        if (successCallback != null) {
+            task.addOnSuccessListener(new OnSuccessListener<String>() {
+                @Override
+                public void onSuccess(String result) {
+                    successCallback.onGameStateSaved(true, null);
+                }
+            }).addOnFailureListener(new OnFailureListener() {
+                @Override
+                public void onFailure(Exception e) {
+                    successCallback.onGameStateSaved(false, e.getMessage());
+                }
+            });
+        }
+    }
+
+    @Override
+    public boolean fetchGameStates(final IFetchGameStatesListResponseListener callback) {
+        if (!this.isSaveDataEnabled) {
```

Comment:
shouldn't be exposed to clients, as there is no way to load another gamestate than the first one